src/main/java/pacman/Ghost.java

```diff
@@ -93,44 +108,62 @@ public boolean isPosibleMoveRight(ArrayList<Position> walls) {
         return true;
     }
 
-    public void searchRoute() {
+    public void searchRoute(Pacman pacman)
+    {
         int dir = new Random().nextInt(4);
 
         switch (dir) {
         case 0:
             if (isPosibleMoveDown(walls)) {
-            moveDown();
+                moveDown();
+                if(existPacmanEatable(pacman)) eatPacman(pacman);
             }
             break;
         case 1:
             if (isPosibleMoveUp(walls)) {
-            moveUp();
+                moveUp();
+                if(existPacmanEatable(pacman)) eatPacman(pacman);
             }
             break;
         case 2:
             if (isPosibleMoveLeft(walls)) {
-            moveLeft();
+                moveLeft();
+                if(existPacmanEatable(pacman)) eatPacman(pacman);
```

Comment:
Please add a single space after "if" and move the code: eatPacman(pacman) to the bottom line